### ISSUE TITLE
Allow the use of C++ exceptions

### ISSUE THREAD

**Issue by tim-lyon-arup** [arup-group/programming-styleguide#1]
The style guide does not allow use of C++ exceptions, but we use them extensively in our codebase, and will probably continue to do so, so would like to update the guide to allow their use.
(https://arup-group.github.io/programming-styleguide/cppguide.html#Exceptions)

**Comment by nathanrobertsarup** [comment]
I agree. https://docs.microsoft.com/en-us/cpp/cpp/errors-and-exception-handling-modern-cpp?view=msvc-170
